src/components/hooks/useWS.ts

```diff
@@ -15,13 +16,29 @@ const useWS = <T extends TSocketEndpointNames>(name: T) => {
     const send = useCallback(
         async (data?: Parameters<typeof apiManager.augmentedSend<T>>[1]) => {
             setIsLoading(true)
-            try {
-                const response = await apiManager.augmentedSend(name, data)
-                setData(response[name] as TSocketResponseData<T>)
-            } catch (e) {
-                setError(e)
-            } finally {
-                setIsLoading(false)
+            const readyState = parseInt(ApiManager.readyState)
+            if (readyState === 1 || readyState === 0) {
+                try {
+                    const response = await apiManager.augmentedSend(name, data)
+                    setData(response[name] as TSocketResponseData<T>)
+                } catch (e) {
+                    setError(e)
+                } finally {
+                    setIsLoading(false)
+                }
+            } else {
+                if (isBrowser()) {
+                    const currentLanguage = getLanguage() ?? 'en'
+                    apiManager
+                        .reconnectIfNotConnected(currentLanguage)
```

Comment:
question: if I understand correctly, if its already connected, it will send using ".augumentedSend", if its not connected already, it will connect and then send using just ".send()". Does that matter / is it accidentall difference / purposefull difference?

also, would it be easier to do slightly differenct logic:

a) check if its connected at the top of the function, if its not, reconnect,
b) just do exactly the same which is doing now, without extra conditions?

Reply:
so something like this 
`    const send = useCallback(
        async (data?: Parameters<typeof apiManager.augmentedSend<T>>[1]) => {
            setIsLoading(true)

            // Check if WebSocket connection is available
            if (parseInt(ApiManager.readyState) !== 1 && parseInt(ApiManager.readyState) !== 0) {
                if (isBrowser()) {
                    const currentLanguage = getLanguage() ?? 'en'
                    try {
                        // Connect to WebSocket if not connected
                        await apiManager.reconnectIfNotConnected(currentLanguage)
                    } catch (e) {
                        setError(e)
                        setIsLoading(false)
                        return
                    }
                } else {
                    setIsLoading(false)
                    return
                }
            }

            // Proceed with sending data
            try {
                const response = await apiManager.augmentedSend(name, data)
                setData(response[name] as TSocketResponseData<T>)
            } catch (e) {
                setError(e)
            } finally {
                setIsLoading(false)
            }
        },
        [name],
    )`

Reply:
something like this 


    const send = useCallback(
        async (data?: Parameters<typeof apiManager.augmentedSend<T>>[1]) => {
            setIsLoading(true)

            // Check if WebSocket connection is available
            if (parseInt(ApiManager.readyState) !== 1 && parseInt(ApiManager.readyState) !== 0) {
                if (isBrowser()) {
                    const currentLanguage = getLanguage() ?? 'en'
                    try {
                        // Connect to WebSocket if not connected
                        await apiManager.reconnectIfNotConnected(currentLanguage)
                    } catch (e) {
                        setError(e)
                        setIsLoading(false)
                        return
                    }
                } else {
                    setIsLoading(false)
                    return
                }
            }

            // Proceed with sending data
            try {
                const response = await apiManager.augmentedSend(name, data)
                setData(response[name] as TSocketResponseData<T>)
            } catch (e) {
                setError(e)
            } finally {
                setIsLoading(false)
            }
        },
        [name],
    )